monkey/monkey_island/cc/ui/src/services/AuthService.js

```diff
@@ -29,16 +34,45 @@ export default class AuthService {
       .then(response => response.json())
       .then(response => {
         if(response.meta.code === 200){
-          this._removeToken();
+          this._removeToken(this.AUTH_TOKEN_NAME_IN_LOCALSTORAGE);
         }
         return response;
       });
   }
 
   authFetch = (url, options) => {
+    this._refreshAuthToken(url, options);
     return this._authFetch(url, options);
   };
 
+  _refreshAuthToken = (url, options) => {
+    this._authFetch(url, options).then(res => {
+      if(res.status == 401)
+      {
+        let refreshToken = this._getToken(this.REFRESH_TOKEN_NAME_IN_LOCALSTORAGE);
+        if (refreshToken !== undefined && refreshToken !== null) {
```

Comment:
Same as writing `if(refreshToken) { // rest of the code }`
I think it will be more understandable (it will return `false` if the 'refreshToken' value is null or undefined.